freqtrade/persistence/models.py

```diff
@@ -505,56 +630,135 @@ def recalc_open_trade_value(self) -> None:
         """
         self.open_trade_value = self._calc_open_trade_value()
 
+    def calculate_interest(self, interest_rate: Optional[float] = None) -> Decimal:
+        """
+        : param interest_rate: interest_charge for borrowing this coin(optional).
+        If interest_rate is not set self.interest_rate will be used
+        """
+        # TODO-mg: Need to set other conditions because sometimes self.open_date is not defined, but why would it ever not be set
```

Comment:
This is most likely only a test problem - in real runs (outside of pytest) - open_date will always be set.